lib/src/test/kotlin/starknet/provider/ProviderTest.kt

```diff
@@ -1200,4 +1200,53 @@ class ProviderTest {
 
         assertNotNull(response)
     }
+
+    @Test
+    fun `batch call contract with block hash and block tag`() {
+        val call1 = Call(
+            contractAddress = balanceContractAddress,
+            entrypoint = "get_balance",
+            calldata = emptyList(),
+        )
+        val blockHash = provider.getBlockHashAndNumber().send().blockHash
+
+        val call2 = Call(
+            contractAddress = balanceContractAddress,
+            entrypoint = "get_balance",
+            calldata = emptyList(),
+        )
+
+        val providerCalls = listOf(
+            provider.callContract(
+                call = call1,
+                blockHash = blockHash,
+            ),
+            provider.callContract(
+                call = call2,
+                blockTag = BlockTag.LATEST,
+            ),
+        )
+        val request = provider.batchRequests(providerCalls)
+        org.junit.jupiter.api.assertAll({ request.send() })
```

Comment:
Let's import this instead.

Reply:
Updated, it was completely removed (explanation in following comment).